flexget/plugins/input/telegram_input.py

```diff
@@ -0,0 +1,228 @@
+                                    'type': 'object',
+                                    'properties': {
+                                        'first': {'type': 'string'},
+                                        'sur': {'type': 'string'},
+                                    },
+                                    'required': ['first', 'sur'],
+                                    'additionalProperties': False,
+                                }
+                            },
+                            'required': ['fullname'],
+                            'additionalProperties': False,
+                        },
+                        {
+                            'type': 'object',
+                            'properties': {'group': {'type': 'string'}},
+                            'required': ['group'],
+                            'additionalProperties': False,
+                        },
+                    ]
+                },
+            },
+
+            'clean_update': {'type': 'boolean','default': True},
+            'entry': {
+                'type': 'object',
+                'properties': {
+                    'url': {'type': 'string', 'format': 'regex'},
+                    'title': {'type': 'string', 'format': 'regex'},
+                },
+                'additionalProperties': {'type': 'string', 'format': 'regex'},
+            },
+        },
+        'required': ['token'],
+        'additonalProperties': False,
+    }
+
+    @plugin.internet(logger)
+    def on_task_input(self, task, config):
+        #Load The Configs
+        token = config['token']
+        clean_update = config['clean_update']
+        entry_config = config.get('entry')
+        whitelist = config.get('whitelist')
+        types = config.get('types')
+
+        #Defaults
+        if not types:
+            types = ['private', 'group']
+
+        #Get Last Checked ID
+        update_id = task.simple_persistence.get(token+'_update_id')
+
+        #Set Last Checked ID as Offset
+        offset = ""
+        if update_id:
+            update_id += 1
+            offset = "?offset="+str(update_id)
+        
+        #The Target URL
+        url = _TELEGRAM_API_URL + "bot" + token + "/getUpdates" + offset  
+
+        #Get Telegram Updates
+        try:
+            response = task.requests.get(
+                url, timeout=60, raise_status=False
+            ).json()
+        except:
+            logger.error("Error getting telegram update")
+            return
+
+        #We have a error
+        if not response['ok']:
+            logger.error("Telegram updater returned error {}: {}", response['error_code'], response['description'] )
+            return
+
+        #Get All New Messages
+        messages = response['result']
+
+        entries = []
+        for message in messages:
+            #This is the ID
+            update_id = message['update_id']
+
+            #If desired update the last ID for the Bot
+            if clean_update:
```

Comment:
im not sure i understand what this option mean

Reply:
I renamed the option to "only_new", this will make it clearer that it only returns new messages, and not processed ones